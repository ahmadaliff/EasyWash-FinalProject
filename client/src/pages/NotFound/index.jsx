import { FormattedMessage } from 'react-intl';
import { useNavigate } from 'react-router-dom';

import logo from '@static/images/notfound.svg';

import classes from './style.module.scss';

const NotFound = () => {
  const navigate = useNavigate();
  return (
    <div className={classes.contentWrapper}>
      <img className={classes.image} src={logo} alt="Not Found" />
      <a href="https://storyset.com/web" className={classes.storySet}>
        Web illustrations by Storyset
      </a>
      <div className={classes.title}>
        <FormattedMessage id="app_not_found" />
      </div>
      <div className={classes.desc}>
        <FormattedMessage id="app_not_found_desc" />
      </div>
      <button className={classes.backButton} type="button" onClick={() => navigate('/')}>
        <FormattedMessage id="app_back_to_home" />
      </button>
    </div>
  );
};

export default NotFound;
