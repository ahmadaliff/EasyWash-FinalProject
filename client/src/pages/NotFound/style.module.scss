@import '@styles/base';

.contentWrapper {
  @include directionY;
  padding: 1rem;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: var(--min-height);
  @include apply-at-md-size {
    min-height: var(--min-height-sm);
  }

  .image {
    width: 20rem;
  }
  .storySet {
    margin: 0;
    font-size: x-small;
    color: var(--color-primary);
  }
  .title {
    margin-top: 0.5rem;
    font-size: 1rem;
    font-weight: 700;
  }
  .desc {
    margin: 0;
    max-width: 20rem;
    text-align: center;
    font-size: medium;
  }
  .backButton {
    background-color: var(--color-primary);
    text-transform: capitalize;
    color: var(--color-bg);
    padding: 0.5rem 1rem;
    border-radius: 10px;
    font-size: medium;
    font-weight: 600;
    cursor: pointer;
    margin: 2rem;
    border: none;
    &:hover {
      transform: scale(1.05, 1.05);
    }
  }
}
