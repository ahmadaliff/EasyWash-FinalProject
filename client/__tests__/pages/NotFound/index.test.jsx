import NotFound from '@pages/NotFound';
import { render, fireEvent, act } from '@utils/testHelper';

let wrapper;
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

beforeEach(() => {
  act(() => {
    wrapper = render(<NotFound />);
  });
});

describe('Not Found Page', () => {
  test('render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('not-found')).toBeInTheDocument();
  });
  test('navigate back when back button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-home');
    await act(async () => {
      fireEvent.click(button);
    });
    expect(mockNavigate).toHaveBeenCalledWith('/');
  });
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
