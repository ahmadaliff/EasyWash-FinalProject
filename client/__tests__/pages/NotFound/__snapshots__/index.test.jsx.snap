// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="contentWrapper"
        data-testid="not-found"
      >
        <img
          alt="Not Found"
          class="image"
        />
        <a
          class="storySet"
          href="https://storyset.com/web"
        >
          Web illustrations by Storyset
        </a>
        <div
          class="title"
        >
          Halaman tidak ditemukan
        </div>
        <div
          class="desc"
        >
          Maaf halaman yang kamu cari gak ada nih, (error 404)
        </div>
        <button
          class="backButton"
          data-testid="button-home"
          type="button"
        >
          Kembali ke halaman utama
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="contentWrapper"
      data-testid="not-found"
    >
      <img
        alt="Not Found"
        class="image"
      />
      <a
        class="storySet"
        href="https://storyset.com/web"
      >
        Web illustrations by Storyset
      </a>
      <div
        class="title"
      >
        Halaman tidak ditemukan
      </div>
      <div
        class="desc"
      >
        Maaf halaman yang kamu cari gak ada nih, (error 404)
      </div>
      <button
        class="backButton"
        data-testid="button-home"
        type="button"
      >
        Kembali ke halaman utama
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
